### ISSUE TITLE
A GigE camera's temperature is always 0.0

### ISSUE THREAD

**Issue by boxanm** [basler/pylon-ros-camera#245]
### Describe what you want to implement and what the issue & the steps to reproduce it are:

I noticed that the `temperature` field of the `/basler/current_params` topic is always 0.0.
After some investigation, I noticed that the [PR](https://github.com/basler/pylon-ros-camera/pull/50) that introduced temperature readings treats GigE and USB cameras differently.
Swapping the original code in the GigE camera call:
```C++
return static_cast<float>(cam_->TemperatureAbs.GetValue());
```
for the USB camera version:
```C++
return static_cast<float>(cam_->DeviceTemperature.GetValue());
```
fixed the issue, and I'm now able to monitor the camera's temperature. I'll be happy to submit a PR with this fix, assuming that the temperature calls in the Pylon SDK have changed since the original PR was submitted. Can someone with more experience with Pylon confirm my hypothesis?

### Hardware setup description

We have a Basler a2A1920-51gcPRO camera. I wasn't able to verify any other device.

### Runtime information

```shell
Ros2 Humble
```

### Is your camera operational with the Basler pylon Viewer on your platform?

Yes

**Comment by SMA2016a** [comment]
Check if your camera has got the temperature sensor with pylonviewer.

**Comment by boxanm** [comment]
Yes, I can get the temperature in PylonViewer, as well as when when using the `cam_->DeviceTemperature.GetValue()` call. Nevertheless, the code sample uses another function call to read it
![Screenshot from 2024-09-23 22-48-36](https://github.com/user-attachments/assets/59c333f3-c26b-4603-9532-45e12ab86c6d)

**Comment by FrancoisPicardDTI** [comment]
Hi @boxanm 
Can you try the latest humble driver version please? It should be fixed.
